app/controllers/api/v0/comments_controller.rb

```diff
@@ -3,21 +3,43 @@ module V0
     class CommentsController < ApiController
       respond_to :json
 
-      caches_action :index,
-                    cache_path: proc { |c| c.params.permit! },
-                    expires_in: 10.minutes
-
-      caches_action :show,
-                    cache_path: proc { |c| c.params.permit! },
-                    expires_in: 10.minutes
+      before_action :set_cache_control_headers, only: %i[index show]
 
       def index
         article = Article.find(params[:a_id])
-        @comments = article.comments.roots.includes(:user).select(%i[id processed_html user_id ancestry])
+
+        @comments = article.comments.includes(:user).select(%i[id processed_html user_id ancestry]).arrange
```

Comment:
here we tell `ancestry` to build the comments in a hash that contains all the tress:

```text
- comment 1
  - reply to comment 1
- comment 2
  - reply to comment 2
    - reply to reply to comment 2
- comment 3
```

It will then be navigated by the `edge_cache_keys` method and the JSON serializer

Reply:
Same comment as below, could we use a joins here?